crates/core/tedge_mapper/src/operations.rs

```diff
@@ -0,0 +1,263 @@
+use std::{
+    collections::HashMap,
+    fs,
+    path::{Path, PathBuf},
+};
+
+use crate::error::OperationsError;
+
+/// Operations are derived by reading files subdirectories per cloud /etc/tedge/operations directory
+/// Each operation is a file name in one of the subdirectories
+/// The file name is the operation name
+
+type Cloud = String;
+type OperationName = String;
+type Operation = HashMap<OperationName, PathBuf>;
+type OperationsMap = HashMap<Cloud, Operation>;
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct Operations {
+    dir: PathBuf,
+    operations: OperationsMap,
+}
+
+impl Default for Operations {
+    fn default() -> Self {
+        Self {
+            dir: Path::new("/etc/tedge/operations").to_path_buf(),
+            operations: HashMap::new(),
+        }
+    }
+}
+
+impl Operations {
+    pub fn new(dir: impl AsRef<Path>) -> Self {
```

Comment:
I would remove this `new()` function as well as the `Default` implementation to keep only the `Operations::try_new()` function.